plugins/uefi-capsule/fu-plugin-uefi-capsule.c

```diff
@@ -433,6 +481,12 @@ fu_plugin_uefi_capsule_load_config (FuPlugin *plugin, FuDevice *device)
 	fu_device_set_metadata_boolean (device,
 					"FallbacktoRemovablePath",
 					fallback_removable_path);
+
+	/* delivery of Capsules via file on Mass Storage device */
+	capsule_on_disk = fu_plugin_get_config_value_boolean (plugin, "DisableCapsuleUpdateOnDisk");
```

Comment:
i'm confused why we'd want to disable it if the platform explicitly says "this is supported" -- is this in case the firmware gets the OS_INDICATIONS value wrong?

Reply:
Because the firmware might support 2 delivery methods.  Runtime capsuleupdates (which is what currently works on fwupd) and capsuleupdate on-disk.  With the way the patch is structured the on-disk takes precedence over the existing method, so a user might want to explicitly disable it

Reply:
Actually I am going to change this, since I don't like having it as a config option.  The runtime capability of the capsule update is described on the RT_PROPERTIES table and even if it's not the QueryCapsuleCapabilities will return EFI_UNSUPPORTED.  So I'll get rid of the config option, try the runtime capsule update first and that is unsupported, we can try the on-disk one

Reply:
> try the runtime capsule update first and that is unsupported, we can try the on-disk one

That doesn't sound very explicit and I think will make it difficult to debug at scale. I would much rather it's explicit.

Reply:
It would involve at least understanding if the firmware suppots a runtime capsule update service. 
That's fine, I changed the logic slightly on the entire patchset.  I think we are safe to keep things as config options for now. 

What I did is: 
1. Make this argument called EnableCapsuleUpdateOnDisk, so the user has to explicitly enable it now and we wont break existing installations
2. The setvariable problem is a bit weird, since U-Boot has no way of letting us know if ignores OsIndications.  So I added another config option for the user to enable 

I understand the goal should be that this works out of the box, but the setvariable problem is weird. I am currently working on it, and I do have a fix, but keep in mind there might be devices running U-Boot that will never support it

Reply:
@hughsie I've tested the whole thing on an RPI4. What's missing from the current code is fixing `fu_uefi_backend_check_efivarfs_rw`, we need to disable that check if setvariable isn't supported at runtime.  In that case the efiavarfs is mount as ro.  
So I was thinking, maybe it's better to rename IgnoreOsIndications to SetVariableMissing or something like that and skip all the calls to it ?

Reply:
> maybe it's better to rename IgnoreOsIndications to SetVariableMissing or something

Ahh, is the IgnoreOsIndications the way of the firmware basically saying "I don't support RT->SetVariable"? Maybe just quirking the "ignore NOT_SUPPORTED" in fwupd would be a quicker approach for now?

Reply:
> > maybe it's better to rename IgnoreOsIndications to SetVariableMissing or something
> 
> Ahh, is the IgnoreOsIndications the way of the firmware basically saying "I don't support RT->SetVariable"? 

Yes, but we only use it to skip setting OsIndications, hence the naming.

> Maybe just quirking the "ignore NOT_SUPPORTED" in fwupd would be a quicker approach for now?

Yea and probably better, since we'll be able to skip setting variables automatically, instead of adding a config option.  I have no idea how to do that though :)